### PR TITLE
Adding SecurityPartnerProvider Resource

### PR DESCRIPTION
**Pull request by ishaniGupta27**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lirenhe** [comment]
@ishaniGupta27, could you take a look at the tool check error? It looks you missed the definition of SecurityPartnerProviderListResult and will you also update the readme to include new file?

**Comment by ishaniGupta27** [comment]
Addressed the comments:

1. Added definition for SecurityPartnerProviderListResult
2.  Updated Readme.md with securityPartnerProvider

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/securityPartnerProviders/{securityPartnerProviderName}": {
+      "delete": {
+        "tags": [
+          "securityPartnerProviders"
+        ],
+        "operationId": "securityPartnerProviders_Delete",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]:
```suggestion
        "operationId": "SecurityPartnerProviders_Delete",
```

Please do this change in other parts of the spec

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/securityPartnerProviders/{securityPartnerProviderName}": {
+      "delete": {
+        "tags": [
+          "securityPartnerProviders"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]:
```suggestion
          "SecurityPartnerProviders"
```

Please do this change in other parts of the spec

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/securityPartnerProviders": {
+      "get": {
+        "tags": [
+          "securityPartnerProviders"
+        ],
+        "operationId": "securityPartnerProviders_List",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]:
```suggestion
        "operationId": "SecurityPartnerProviders_ListByResourceGroup",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/securityPartnerProviders": {
+      "get": {
+        "tags": [
+          "securityPartnerProviders"
+        ],
+        "operationId": "securityPartnerProviders_ListAll",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]:
```suggestion
        "operationId": "SecurityPartnerProviders_List",
```

**Review by anton-evseev — CHANGES_REQUESTED** [review]
LGTM except some nitpicks

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json (hunk 1 above)
Comment (ishaniGupta27) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]: Done..

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json (hunk 3 above)
Comment (ishaniGupta27) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]: Done..

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json (hunk 4 above)
Comment (ishaniGupta27) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]: Done..

**Review by lirenhe — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json (hunk 2 above)
Comment (ishaniGupta27) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/securityPartnerProvider.json]: Done.

**Review by anton-evseev — APPROVED** [review]

**Pull request merged**